app/services/auth/auth_info_params_builder.rb

```diff
@@ -24,8 +24,9 @@ def call
     def load_email(params)
       return if params[:email].present?
 
+      # TODO: why do we need USER_API_CLIENT? GitLab always have an email and there is no :email method in ProviderAPI::Gitlab::UserClient
```

Comment:
we have to ensure gitlab always has a user email, if it is, I agree to remove `USER_API_CLIENT`

Reply:
https://github.com/CleverLabs/deployqa/issues/259